packages/model-viewer/src/features/scene-graph/material.ts

```diff
@@ -66,57 +66,69 @@ export class Material extends ThreeDOMElement implements MaterialInterface {
 
     let {normalTexture, occlusionTexture, emissiveTexture} = gltfMaterial;
 
-    const normalTextures = new Set<ThreeTexture>();
-    const occlusionTextures = new Set<ThreeTexture>();
-    const emissiveTextures = new Set<ThreeTexture>();
+    let normalTextures: ThreeTexture|null = null;
+    let occlusionTextures: ThreeTexture|null = null;
+    let emissiveTextures: ThreeTexture|null = null;
 
-    for (const gltfMaterial of correlatedMaterials) {
-      const {normalMap, aoMap, emissiveMap} = gltfMaterial;
+    const {normalMap, aoMap, emissiveMap} =
+        correlatedMaterials.values().next().value;
 
-      if (normalTexture != null && normalMap != null) {
-        normalTextures.add(normalMap);
-      } else {
-        normalTexture = {index: -1};
-      }
+    if (normalTexture != null && normalMap != null) {
+      normalTextures = normalMap;
+    } else {
+      normalTexture = {index: -1};
+    }
 
-      if (occlusionTexture != null && aoMap != null) {
-        occlusionTextures.add(aoMap);
-      } else {
-        occlusionTexture = {index: -1};
-      }
+    if (occlusionTexture != null && aoMap != null) {
+      occlusionTextures = aoMap;
+    } else {
+      occlusionTexture = {index: -1};
+    }
 
-      if (emissiveTexture != null && emissiveMap != null) {
-        emissiveTextures.add(emissiveMap);
-      } else {
-        emissiveTexture = {index: -1};
-      }
+    if (emissiveTexture != null && emissiveMap != null) {
+      emissiveTextures = emissiveMap;
+    } else {
+      emissiveTexture = {index: -1};
     }
 
-    const firstValue = (set: Set<ThreeTexture>): ThreeTexture => {
-      return set.values().next().value;
-    };
+    for (const gltfMaterial of correlatedMaterials) {
+      const {
+        normalMap: verifyNormalMap,
+        aoMap: verifyAoMap,
+        emissiveMap: verifyEmissiveMap
+      } = gltfMaterial;
+      if (verifyNormalMap !== normalMap) {
+        console.error('Normal map differs between homegenous materials');
+      }
+      if (verifyAoMap !== aoMap) {
+        console.error('AO map differs between homegenous materials');
+      }
+      if (verifyEmissiveMap !== emissiveMap) {
+        console.error('Emissive map differs between homegenous materials');
+      }
+    }
 
     this[$normalTexture] = new TextureInfo(
         onUpdate,
         gltf,
-        this[$backingThreeMaterial],
-        firstValue(normalTextures),
+        correlatedMaterials,
+        normalTextures,
         TextureUsage.Normal,
         normalTexture!);
 
     this[$occlusionTexture] = new TextureInfo(
         onUpdate,
         gltf,
-        this[$backingThreeMaterial],
-        firstValue(occlusionTextures),
+        correlatedMaterials,
```

Comment:
The main thing that changes here is that all correlated materials are now being being updated when editing a texture. However, I simplified things a bit be removing, what ends up being, duplicate references to the same textures. So when a material is branched in Three.js, because of some state change, both branches still  point to the same texture, any request to change the texture affects all materials but there is no need to represent the same texture many times.